src/test/java/automation/tests/ManagingGroupTest.java

```diff
@@ -0,0 +1,130 @@
+package automation.tests;
+
+import automation.pages.BrowseGroupsPanel;
+import automation.pages.LoginPage;
+import automation.pages.NewGroupPage;
+import automation.pages.UpdateGroupPage;
+import automation.utils.WaitForElement;
+import automation.utils.loaders.EnvironmentConfigLoader;
+import automation.utils.loaders.Go;
+import automation.utils.loaders.Pages;
+import automation.utils.loaders.UserConfigLoader;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.openqa.selenium.WebDriver;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DisplayName("TS01 - Managing groups")
+public class ManagingGroupTest {
+
+    private static Go go;
+    private static WebDriver driver;
+    private BrowseGroupsPanel browseGroupsPanel;
+
+    @DisplayName("TC01 - New group can be created")
+    @ParameterizedTest
+    @MethodSource("groupCredentialsProvider")
+    void checkIfNewGroupCanBeCreated(final String identifier, final String displayName) {
+        //given:
+        go.to(Pages.NEW_GROUP_PAGE);
+
+        //when:
+        NewGroupPage newGroupPage = new NewGroupPage(driver);
+        newGroupPage.typeIdentifier(identifier);
+        newGroupPage.typeDisplayName(displayName);
+        newGroupPage.submitCreateGroupButton();
+
+        browseGroupsPanel = new BrowseGroupsPanel(driver);
+        browseGroupsPanel.setPathToGroupCredentials(displayName, identifier);
+
+        WaitForElement.wait(driver, browseGroupsPanel.getGroupCredentials());
+
+        //then
+        assertEquals(String.format("%s (%s)", displayName, identifier), browseGroupsPanel.getGroupName(),
+                "Group 'display name' and/or 'identifier' are not the same");
+    }
+
+    @DisplayName("TC02 - Existing group can be edited")
+    @ParameterizedTest
+    @MethodSource("groupDisplayNameEditText")
+    void checkIfGroupCanBeEdited(final String displayName, final String identifier, final String editText) {
+        //given:
+        // todo: do it in the way which allow to put group name dynamically
+        go.to(Pages.SOME_GROUP_EDIT_PAGE);
+
+        //when:
+        UpdateGroupPage updateGroupPage = new UpdateGroupPage(driver);
+
+        WaitForElement.wait(driver, updateGroupPage.getEditGroupInputField());
+
+        updateGroupPage.editGroupName(editText);
+        updateGroupPage.clickUpdateButton();
+
+        browseGroupsPanel = new BrowseGroupsPanel(driver);
+        browseGroupsPanel.setPathToGroupCredentials(String.format("%s%s", displayName, editText), identifier);
+
+        WaitForElement.wait(driver,browseGroupsPanel.getGroupCredentials());
+
+        //then:
+        assertEquals(String.format("%s%s (%s)", displayName, editText, identifier), browseGroupsPanel.getGroupName(),
```

Comment:
Do not repeat yourself. You are forced to copy-paste your tests because you decomposed task wrong. Use pageObject pattern to perform business tasks (e.g. page.createGroup(groupName, groupDescription) instead of page.putGroupNameText(), page.putGroupNameDesc(), page.clickSubmit()...). If there is 30 clicks to delete group, you are going to call 30 methods in each test?

Reply:
Understood